### PR TITLE
Adding `percentage` metric to aggregations.

### LINKED ISSUE #6763: Percentage in views data table
We'd like to be able to show the percentage each row of a views data table contributes to a total, which might be `count()` or something like `sum(traffic_accounting_size)`.

## Expected Behavior

As is possible with the old search page's "Quick values" and the "Pie Chart" visualization, we'd like to be able to show the percentage a particular row contributes to a certain total.

## Current Behavior

It looks like the pie chart might be the only visualization that shows percentages.

## Possible Solution

A `percentage(...)` metric that can be used to add a data table column representing the percentage a given row contributes to a certain sum.

## Context

We're putting together a view to monitor `traffic_accounting_size`, `count()`, etc. per log source.  It'd be useful to have data table columns representing the various percentages each source contributes to various totals.

## Your Environment

* Graylog Version: 3.1.3
* Elasticsearch Version: 6.6.2-1
* Operating System: RHEL

### LINKED ISSUE #7822: Provide percentage and count view just like "quick values" of Graylog Version pre 3.2
Beginning with Graylog Version 3.2, I am missing a daily used feature

**Pre Vers 3.2 had "quick values**"

quick value show both:
a) percentage and
b) count
of a field value

I really felt in love with this table having percentage and count value in one row.
Number one usage for investigations!
grafik

**since Vers 3.2.3 "show top values"**
show top values shows only:

a) count
of a field value

Now you need to create another widget (Pie chart) to get precentage
grafik

It’s getting more complicated and the “percentage” values of field names are getting harder to see.

A table view (message table) including “count” and “percentage” was perfect!

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is implementing a `percentage` metric, which retrieves a value between 0 and 1 to reflect the relative share of the current bucket in relation to the total set of documents. It allows configuring two strategies:

  - `count` (default): Returns the relative share of documents in this bucket compared to the total set of documents. Takes an optional field, which would be used to count only documents containing this field.
  - `sum`: Takes a mandatory field name and returns the relative sum of all values of this field in this bucket compared to the total sum of all field values in the total set of documents.

image

The total set of documents depends on the aggregation configured. When only row pivots are configured, the total set of documents is what the current time range/query returns. When column pivots are configured, the total set of documents is the set of documents contained in the buckets returned by row pivots. This allows e.g. returning the relative distribution of column pivots over time.

image

In addition, this PR is adding a `percentage` metric automatically for the aggregation generated by the `Show Top Values` action.

Fixes #6763.
Fixes #7822.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/aggregationwizard/metric/MetricConfiguration.tsx [hunk 1]
```diff
@@ -129,6 +138,46 @@ const Metric = ({ index }: Props) => {
...
+                     labelClassName="col-sm-3"
+                     wrapperClassName="col-sm-9">
+                <FieldSelect id="metric-field-select"
+                             selectRef={metricFieldSelectRef}
+                             onChange={(fieldName) => onChange({ target: { name, value: fieldName } })}
+                             clearable={!isFieldRequired}
+                             properties={percentageRequiredFieldOptions}
+                             name={name}
+                             value={value}
+                             ariaLabel="Select a field" />
```
[the remark this answers is not in the document]
Comment (linuspahl) [reply graylog2-web-interface/src/views/components/aggregationwizard/metric/MetricConfiguration.tsx:175]: Let's make this field required, when the strategy is `Feld Sum`. Currently the search fails, when there is no field defined in this case and the form state seems to stay "dirty".

test

**Review by linuspahl — COMMENTED** [review]
Tested different use cases and in general it works very well!

**Review by linuspahl — APPROVED** [review]
Frontend and functionality look good to me!

**Review by janheise — APPROVED** [review]
LGTM! 🎉

**Pull request merged**